### PR TITLE
default_coafile: Rename to system_coafile

### PR DESCRIPTION
**Pull request by acashmkj**
Renamed default_coafile to system_coafile and its
occurence in Constants.py and setup.py

Closes https://github.com/coala/coala/issues/3389

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by RaiVaibhav** [comment]
@acashmkj  does you renamed or deleted last and created new???

**Comment by acashmkj** [comment]
@RaiVaibhav deleted default_coafile and them created a new system_coafile

**Comment by RaiVaibhav** [comment]
@acashmkj  I think when you deletion effects the `cache` and helps it in run

**Comment by acashmkj** [comment]
@RaiVaibhav yup. I totally forgot about the cache.

**Comment by Makman2** [comment]
Please read https://github.com/coala/coala/pull/4763#issuecomment-337632934 again carefully, currently you introduce an API break.

**Comment by Makman2** [comment]
unack 344bd61

**Comment by RaiVaibhav** [comment]
@makman2 can you be more specific i did't get it how does API break here

**Comment by Makman2** [comment]
Users can currently rely on the filename `default_coafile` when they want to use system-wide settings. If we just change it to `system_coafile`, a person's `default_coafile` on his system won't be recognized by coala any more.

**Comment by RaiVaibhav** [comment]
ok I got it the `dev` i.e the version we install using `pip` will contain the `default_coafile` file but after changes i will cause error as `default_coafile` is not there but what we can do is deprecate the `default_coafile` and add discription about this changes , am i right @Makman2 ??

**Comment by RaiVaibhav** [comment]
unack 344bd6141e2572b66a75bfdac14e186fa9e89b09

**Comment by Makman2** [comment]
No it won't error, but we just change the filename we tell our users to use for this. This means we change public API, so we have to deprecate things properly.

File: tests/settings/ConfigurationGatheringTest.py [hunk 1]
```diff
@@ -101,13 +101,50 @@ def test_default_coafile_parsing(self):
...
+        tmp = Constants.system_coafile
+
+        Constants.system_coafile = os.path.abspath(os.path.join(
+            os.path.dirname(os.path.realpath(__file__)),
+            'section_manager_test_files',
+            'default_coafile'))
+
+        logger = logging.getLogger(__name__)
+        with self.assertLogs(logger, 'WARNING') as log:
+            self.assertEqual(log.output,
```
Comment (ayush1999) [inline tests/settings/ConfigurationGatheringTest.py]: Checks are giving `AssertionError` here. Please check.

File: coalib/misc/Constants.py [hunk 2]
```diff
@@ -109,7 +110,16 @@
...
 
-system_coafile = os.path.join(coalib_root, 'default_coafile')
+# system_coafile = os.path.join(coalib_root, 'system_coafile')
+for file_lists in os.listdir(coalib_root):
+    for file in file_lists:
+        if file == 'default_coafile':
+            system_coafile = os.path.join(coalib_root, 'default_coafile')
+        else:
+            system_coafile = os.path.join(coalib_root, 'system_coafile')
+
```
Comment (alphadose) [inline coalib/misc/Constants.py]: You can use a more compact logic to achieve the above result.
file = [i for i in os.listdir(coalib_root) if "coafile" in i][0]
system_coafile = os.path.join(coalib_root, file)

**Comment by alphadose** [comment]
And also read the travis logs for error in tests

**Comment by alphadose** [comment]
@acashmkj Also squash your commits into a single one :+1:

File: tests/settings/ConfigurationGatheringTest.py [hunk 3]
```diff
@@ -101,13 +101,50 @@ def test_default_coafile_parsing(self):
...
+        """
+        tmp = Constants.system_coafile
+
+        Constants.system_coafile = os.path.abspath(os.path.join(
+            os.path.dirname(os.path.realpath(__file__)),
+            'section_manager_test_files',
+            'default_coafile'))
+
+        logger = logging.getLogger(__name__)
+        with self.assertLogs(logger, 'WARNING') as log:
```
Comment (RaiVaibhav) [inline tests/settings/ConfigurationGatheringTest.py]: Here `assertLog(boolean assertion,String msg)`in If assertion parameter is false, then logs message as an error statement. Here ` String` will `Log a message object with the ERROR Level`. and you can check [level](https://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/Level.html#ERROR) 
Here you will find there is not any `level-WARNING`

Here you can [check](https://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/Category.html)

File: tests/settings/ConfigurationGatheringTest.py [hunk 4]
```diff
@@ -101,13 +101,50 @@ def test_default_coafile_parsing(self):
...
+            self.assertEqual(log.output,
+                             ["WARNING: root: Use of file named\
```
Comment (RaiVaibhav) [inline tests/settings/ConfigurationGatheringTest.py]: Please replace `"` with `'`

Comment (Makman2) [reply tests/settings/ConfigurationGatheringTest.py]: and please don't use backslash continuation^^

**Comment by Makman2** [comment]
Closing due to inactivity.

**Pull request closed without merging**